src/content/basics/ha-masters/index.md

```diff
@@ -0,0 +1,169 @@
+
+## Via the Rest API {#rest-api}
+
+Check the [v5 cluster modification API reference](/api/#operation/modifyClusterV5) to find out how to convert a cluster programmatically using the Rest API.
+
+## Via the Control Plane K8s API {#cp-k8s-api}
+
+Two resources have to be changed in order to convert a single master cluster
+to high availability.
+
+1. `AWSControlPlane`
+2. `G8sControlPlane`
+
+### `AWSControlPlane`
+
+The cluster's [`AWSControlPlane`](/reference/cp-k8s-api/awscontrolplanes.infrastructure.giantswarm.io/)
+has a `.spec.availabilityZones` array which, in case of a single master
+cluster, has only one item. This item is the availability zone assigned to the
+only master node.
+
+To convert the cluster to high availability, this array has to be extended
+to have three items. Each item specifies the availability zone for one master
+node after the conversion.
+
+**CAUTION:** Order is important here. The first item must be left untouched. Or, in other words, what was the only array item before must be the first of three item after the change.
+
+Example for a cluster in region `eu-central-1`:
+
+The `AWSControlPlane` shows this spec (shortened):
+
+```yaml
+spec:
+  availabilityZones:
+  - eu-central-1c
+```
+
+The resource gets changed to this:
+
+```yaml
+spec:
+  availabilityZones:
+  - eu-central-1c
+  - eu-central-1a
+  - eu-central-1b
+```
+
+### `G8sControlPlane`
+
+The [`G8sControlPlane`](/reference/cp-k8s-api/g8scontrolplanes.infrastructure.giantswarm.io/)
+has a `.spec.replicas` attribute which indicates the number of master nodes to
+run. This is `1` for a single master cluster and needs to be modified to `3`.
+
+**Note:** Modifying the `G8sControlPlane` resource must happen _after_ the `AWSControlPlane`
+modification in order to have full control over AZ usage. Otherwise the `AWSControlPlane` will
+be modified automatically and AZs will be added by our automation. This may or may not
+be your desired way.
+
+## Technical details
+
+### Why three master nodes
+
+At Giant Swarm we operate "stacked" clusters where etcd nodes are running on the same machines
+as the Kubernetes master nodes. Each master node runs control plane components and a member of
+an etcd cluster.
+
+In order to achieve high availability in an etcd cluster, a quorum (majority of members) is
+needed. With three members, one member is allowed to become dysfunctional at any time, e. g.
+through a machine failure or availability zone outage.
+
+The number of three masters running three etcd members is a good balance between resilience
+and cost. While it would be technically possible to run five or more master nodes, this
+is not currently supported with Giant Swarm tenant clusters.
+
+## Limitations
+
+- Currently only supported AWS, not on Azure and KVM.
+- API downtimes are still possible during cluster modifications, especially when the leader of the
```

Comment:
I would like to add that the disruption is **brief** and **negligible** when using backoff strategies client side.

Reply:
Done